packages/form/src/reatomForm.ts

```diff
@@ -1,176 +1,684 @@
+        field.keepErrorDuringValidating(keepErrorDuringValidating);
+
+      if(field.keepErrorOnChange() === undefined)
+        field.keepErrorOnChange(keepErrorOnChange);
+
+			if (schema) {
+        field.shouldValidate(true);
+
+        field.validation.trigger.extend(
+          withCallHook(() => {
+            if (!field.validation().error && !isCausedBy(submit))
+              checkSchemaValidation(field);
+          })
+        )
+			}
+		},
+  });
+  
+  const {
+    fieldsState,
+    focus,
+    validation
+  } = reatomFieldsSet(fields, name);
+
+  const fieldsList = computed(() => computeFieldsList(fields), `${name}.fieldsList`);
+  const fieldArraysList = computed(() => computeFieldArraysList(fields), `${name}.fieldArraysList`);
+
+  const submitted = atom(false, `${name}.submitted`);
+
+  const reset = action((initState?: FormPartialState<T>) => {
+    if (initState)
+      reinitState(initState, fields);
+
+    fieldArraysList().forEach((fieldArray) => fieldArray.reset());
+    fieldsList().forEach((fieldAtom) => fieldAtom.reset());
+
+    submitted(false);
+    submit.error.reset();
+
+    if(!isCausedBy(submit))
+      submit.abort(`${name}.reset`);
+  }, `${name}.reset`);
+
+  const reinitState = (initState: FormPartialState<T>, fields: FormFields) => {
+    for (const [key, value] of Object.entries(initState as Rec)) {
+      if (isLinkedListAtom(fields[key])) {
+        // @ts-expect-error bad type for initiate
+        fields[key].initState(fields[key].initiateFromSnapshot(value.map(v => [v])));
+      }
+      else if (
+        isObject(value) &&
+        !(value instanceof Date) &&
+        key in fields &&
+        !isAtom(fields[key])
+      ) {
+        reinitState(value, fields[key] as unknown as FormFields);
+      }
+      else if (isAtom(fields[key])) {
+        fields[key].initState(value);
+      }
+    }
+  };
+
+  const init = action((initState: FormPartialState<T>) => {
+    reinitState(initState, fields);
+  }, `${name}.init`);
+
+	const checkSchemaValidation = action(async (triggerOnlyFor?: Atom) => {
+    if (!schema)
+			throw new Error('Triggering schema validation without schema');
+    
+    const state = fieldsState();
+		const validation = schema['~standard'].validate(state);
+		const result = validation instanceof Promise ? await wrap(validation) : validation;
+
+		if (result.issues?.length) {
+			for (const issue of result.issues) {
+				const field = resolveFieldByPath(issue.path, fields);
+				if (!field || (triggerOnlyFor && triggerOnlyFor !== field))
+					continue;
+
+				field.validation.merge({
+					error: issue.message,
+					meta: undefined,
+					triggered: true,
+					validating: false,
+				});
+
+				if (triggerOnlyFor)
+					break;
+			}
+		}
+
+		return result;
+	}, `${name}.checkSchemaValidation`);
+
+  const submit = action(async () => {
+    for (const field of fieldsList()) {
+      if (!field.validation().triggered)
+        field.validation.trigger();
+    }
+
+    notify();
```

Comment:
Do not use notify directly, it is an escape hatches for tests mostly, it didn't call pending microtasks from a user code, use noop await instead